client/my-sites/email/form/mailboxes/validators.ts

```diff
@@ -28,108 +43,232 @@ class RequiredValidator< T > implements Validator< T > {
+class PasswordValidator extends BaseValidator< string > {
+	private readonly minimumPasswordLength: number;
+	static readonly maximumPasswordLength = 100;
+
+	constructor( minimumPasswordLength: number ) {
+		super();
+		this.minimumPasswordLength = minimumPasswordLength;
+	}
+
+	static getPasswordTooShortError( minimumPasswordLength: number ): FieldError {
+		return i18n.translate( "This field can't be shorter than %s characters.", {
+			args: String( minimumPasswordLength ),
+		} );
+	}
+
+	static getPasswordTooLongError(): FieldError {
+		return i18n.translate( "This field can't be longer than %s characters.", {
+			args: PasswordValidator.maximumPasswordLength,
+		} );
+	}
+
+	static getPasswordStartsWithSpaceError(): FieldError {
+		return i18n.translate( "This field can't start with a white space." );
+	}
+
+	static getPasswordEndsWithSpaceError(): FieldError {
+		return i18n.translate( "This field can't end with a white space." );
+	}
+
+	static getPasswordContainsForbiddenCharacterError(
+		firstForbiddenCharacter: string | undefined
+	): FieldError {
+		return i18n.translate( "This field can't accept '%s' as character.", {
+			args: firstForbiddenCharacter,
+			comment: '%s denotes a single character that is not allowed in this field',
+		} );
+	}
+
+	validateField( field: MailboxFormFieldBase< string > ): void {
+		const value = field.value;
+
+		if ( this.minimumPasswordLength > value.length ) {
+			field.error = PasswordValidator.getPasswordTooShortError( this.minimumPasswordLength );
+			return;
+		}
+
+		if ( PasswordValidator.maximumPasswordLength < value.length ) {
+			field.error = PasswordValidator.getPasswordTooLongError();
 			return;
 		}
 
-		const { error } = validatePasswordField( { value: field.value, error: field.error }, 10 );
-		field.error = error;
+		if ( value.startsWith( ' ' ) ) {
+			field.error = PasswordValidator.getPasswordStartsWithSpaceError();
+			return;
+		}
+
+		// Checks that passwords only have ASCII characters (see https://en.wikipedia.org/wiki/ASCII#Character_set)
+		const regexp = /[^\x20-\x7E]/;
+
+		if ( regexp.test( value ) ) {
+			const firstForbiddenCharacter = [ ...value ].find( ( character ) =>
+				regexp.test( character )
+			);
+
+			field.error =
+				PasswordValidator.getPasswordContainsForbiddenCharacterError( firstForbiddenCharacter );
+			return;
+		}
+
+		if ( value.endsWith( ' ' ) ) {
+			field.error = PasswordValidator.getPasswordEndsWithSpaceError();
+		}
 	}
 }
 
-class ExistingMailboxesValidator< T extends EmailProvider > implements Validator< string > {
-	form: MailboxForm< T >;
+class ExistingMailboxNamesValidator extends BaseValidator< string > {
+	existingMailboxNames: string[];
+
+	constructor( existingMailboxNames: string[] ) {
+		super();
+		this.existingMailboxNames = existingMailboxNames;
+	}
 
-	constructor( form: MailboxForm< T > ) {
-		this.form = form;
+	static getExistingMailboxError(): FieldError {
+		return i18n.translate( 'Please use unique mailboxes' );
```

Comment:
Note we should end error messages with a dot:

```suggestion
		return i18n.translate( 'Please use unique mailboxes.' );
```